pkg/apis/internal/gs/state.go

```diff
@@ -0,0 +1,74 @@
+package gs
+
+import (
+	"encoding/json"
+
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// State represents the state object available on GS resources
+type State struct {
+	// programatically usable enum value
+	ID string `json:"id"`
+
+	// human readable status text
+	Text string `json:"text"`
+
+	Type int `json:"type"`
+}
+
+type objectWithStateRetriever interface {
+	types.Object
+	StateRetriever
+}
+
+// StateRetriever is an interface Objects can implement to provide unified state information
+type StateRetriever interface {
+	StateOK() bool
+	StatePending() bool
+	StateError() bool
+}
+
+const (
+	// StateTypeError is used for states of type error
+	StateTypeError = iota
+	// StateTypeOK is used for states of type OK
+	StateTypeOK
+	// StateTypePending is used for states of type Pending
+	StateTypePending
```

Comment:
I'm not sure using `iota` is a good idea here, at least needs a comment to clarify those IDs are defined by the Engine and not us here

Reply:
still lacks a comment saying those IDs are defined by the Engine and not go-anxcloud